nautobot/docs/user-guide/platform-functionality/dynamicgroup.md

```diff
@@ -2,69 +2,107 @@
 
 +++ 1.3.0
 
-Dynamic Groups provide a way to organize objects of the same Content Type by matching filters. A Dynamic Group can be used to create unique groups of objects matching a given filter, such as Devices for a specific location or set of locations. As indicated by the name, Dynamic Groups update in real time as potential member objects are created, updated, or deleted. They are a counterpart to [Static Groups](staticgroup.md), the difference betwen the two being that members of a Static Group are manually (statically) assigned to that group, whereas members of a Dynamic Group are automatically (dynamically) determined based on a set of filter rules.
+Dynamic Groups provide a general-purpose way to collect related objects of a given Content Type, such as Devices associated with a given Location(s) and Status(es), Circuits attached to a given set of Devices, and so forth. A given Dynamic Group may contain any number of objects of the appropriate type as members, and an object may belong to any number of Dynamic Groups of the appropriate content-type as a member of each. It is possible both to query the members of a given Dynamic Group, and also to query which Dynamic Groups a given object is a member of.
 
-When creating a Dynamic Group, one must select a Content Type to which it is associated, for example `dcim.device`. The filtering parameters saved to the group behave as a bi-directional search query that is used to identify members of that group, and can also be used to determine from an individual object the list of Dynamic Groups to which an individual object belongs.
+There are currently three ways to define the object members of a Dynamic group:
 
-Once created the Content Type for a Dynamic Group may not be modified as this relationship is tightly-coupled to the available filtering parameters. All other fields may be updated at any time.
+1. Filter-based assignment (for example, "All Devices with Location X and Status Y")
+2. Set-based assignment from other defined groups (for example, "All devices in Group A or Group B but not Group C")
+3. Static assignment directly to the group (for example, "Device XYZ01, Device XYZ02, and Device XYZ04")
+
++++ 1.4.0 "Set-based group assignment"
+    Support for set-based assignment to groups was added.
+
++++ 2.3.0 "Static group assignment"
+    Support for static assignment to groups was added, and the distinction between filter-based and set-based assignment was formally established.
+
+The members of each Dynamic Group are cached and tracked by [Static Group Association](staticgroupassociation.md) database records.
 
 ## Introduction
 
-### Creating Dynamic Groups
+### Creating Dynamic Groups in the UI
 
-Dynamic Groups can be created through the UI under _Organization > Dynamic Groups_ and clicking the "Add" button, or through the REST API.
+Dynamic Groups can be created through the UI under _Organization > Groups > Dynamic Groups_ and clicking the "Add" button, or through the REST API.
```

Comment:
Nitpick: `in the UI` in the heading and `or through the REST API` in the first line feels weird

Reply:
Agreed, will reword.